File: Data Structures & Algorithms/binary-search/submission-1.java
class Solution {
    public int search(int[] array, int target) {
        // int index = Arrays.binarySearch(nums, target);
        // if (index <= -1){
        //     return -1;
        // }
        // return index;

        int high = array.length - 1;
        int low = 0;
        while((high - low) >= 0){
            int middle = low + ((high - low) / 2);
            if(array[middle] == target){
                return middle;
            }
            else if(target > array[middle]){
                low = middle + 1;
            } else if(target < array[middle]){
                high = middle - 1;
            }
        }
        return -1; //default not found value.
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        my_map = {}

        for i in range(len(nums)):
            dif = target - nums[i]
            if dif in my_map:
                return [my_map[dif], i]
            else:
                my_map[nums[i]] = i

            
        